### ISSUE TITLE
Memory consumption problem

### ISSUE THREAD

**Issue by default1360** [puppet-meteor/EMS#2]
When I run EMS on libpng and libxml, it will occupy 20+ G of memory. I don't know if this is normal. By comparison, AFL++ only occupy 100+ MB.

**Comment by puppet-meteor** [comment]
Hi, in my environment EMS takes about 2G of memory. I'm not sure about your situation, do you mean that EMS takes 20+ G of memory and is still running? Does memory usage increase gradually?

**Comment by default1360** [comment]
Thank you very much for your reply.
The command line I use looks like this: `./afl-fuzz -i ./corpus/ -o ./findings -G ./ems4.txt -L 0 -t 600+ -m 5000 -- ./xmllint --valid --oldxml10 --push --memory @@`.
EMS takes 20+ G of memory and is still running and the memory usage increase gradually.
I don't know if I made a mistake somewhere.

**Comment by puppet-meteor** [comment]
Looks like a code implementation problem, let me check for it.

**Comment by default1360** [comment]
In addition, my test is performed on MAGMA(https://hexhive.epfl.ch/magma/). EMS has taken 4+ G of memory after fuzzing 10 mins on libpng.

**Comment by puppet-meteor** [comment]
Hi, I've used EMS to fuzz xmllint on my server. EMS takes 12G when running for 40+ min. Then, its memory overhead suddenly drops to 2G and increases slowly. I assume the reason is as follows. Before EMS invokes `dict_incremental()` to handle history data, EMS needs to store temporary data in the memory. Thus, if EMS finds plenty of new interesting test cases in a period, it will take a lot of memory. 

Therefore, a naive way to alleviate this phenomenon is to shorten the scheduling period of `dict_incremental()`. In the new commit of EMS, I've adjusted the hyper-parameter `swarm_num` and `period_core` to achieve the above goal. 

BTW, the old version of EMS costs much less memory compared to the open-sourced one, which may be caused by my new designs.
